src/ess/diffraction/grouping.py

```diff
@@ -22,4 +34,48 @@ def group_by_two_theta(data: sc.DataArray, *, edges: sc.Variable) -> sc.DataArra
         `data` grouped into two_theta bins.
     """
     out = data.transform_coords('two_theta', graph=beamline.beamline(scatter=True))
-    return out.bin(two_theta=edges.to(unit=out.coords['two_theta'].unit, copy=False))
+    return MergedPixels[RunType](
+        out.bin(two_theta=edges.to(unit=out.coords['two_theta'].unit, copy=False))
+    )
+
+
+def merge_all_pixels(data: DspacingData[RunType]) -> MergedPixels[RunType]:
+    """Combine all pixels (spectra) of the detector.
+
+    Parameters
+    ----------
+    data:
+        Input data with a `'spectrum'` dimension.
+
+    Returns
+    -------
+    :
+        The input without a `'spectrum'` dimension.
+    """
+    return MergedPixels(data.bins.concat('spectrum'))
+
+
+def finalize_histogram(
+    data: NormalizedByVanadium, edges: DspacingBins
+) -> DspacingHistogram:
```

Comment:
This looks like we are normalizing in event mode, which is bad for error handling. Should histogram before vanadium division?

Reply:
Maybe. I didn't want to alter the workflow.

Reply:
Can you open an issue at least?

Reply:
Done